### PR TITLE
Check sum

### PR DESCRIPTION
**Pull request by tmargary**
- Fixed the libzippp string error on Windows
- Added constness 
- Fixed miscellaneous Windows related issues
- Cleaned the code
- Removed boost packages
- Removed zlib and libzip

### CODE REVIEW

**Review by Mher594 — COMMENTED** [review]
After merging let's delete old branches and not reuse them.

File: external/vcpkg
[the remark this answers is not in the document]
Comment (tmargary) [reply external/vcpkg:1]: Yes. Reverted back to normal.

File: .gitignore [hunk 1]
```diff
@@ -33,8 +33,8 @@
 
 *.user
 /build
+/external
```
[the remark this answers is not in the document]
Comment (tmargary) [reply .gitignore:36]: Noted

File: src/MainView/MainView.cpp [hunk 2]
```diff
@@ -12,7 +12,7 @@ MainView::MainView(QWidget* parent): QWidget(parent) {
     button->setStyleSheet("background-color:green;");
     layout->addWidget(button);
     QObject::connect(button, &QPushButton::clicked, [=](){
-        QString fileName = QFileDialog::getOpenFileName(nullptr, "Open File", QDir::homePath());
+        const QString fileName = QFileDialog::getOpenFileName(nullptr, "Open File", QDir::homePath());
```
[the remark this answers is not in the document]
Comment (tmargary) [reply src/MainView/MainView.cpp:15]: Noted

**Review by Mher594 — APPROVED** [review]

**Review by Gharibayan — APPROVED** [review]

**Pull request merged**